### ISSUE TITLE
Local Geary loses order

### ISSUE THREAD

**Issue by Michel-Anton** [pysal/esda#192]
When running:
```
local_geary = Geary_Local(spatial_weights)
local_geary_contiguity_ratio = local_geary.fit(x)
```
The order of values in `local_geary_contiguity_ratio` does not correspond to the order of values in the input array `x`.

The loss of order occurs at line 167 in local_geary.py:
`adj_list_gs = adj_list_gs.groupby(by="ID").sum()`

The _groupby_ function returns the values in the lexicographic order of the weight id's. While the spatial weights class W stores data in lexicographic order by default, a user may impose a different ordering by setting the _id_order_ parameter. In this case, the order in the _localG_ attribute is different from the input, which is quite misleading and frankly speaking a bug.

It would be quite useful for the _fit_ function to return the values in same order as the spatial weights. If you chose not to do it, please indicate in the documentation that the order of values in the _localG_ attribute may change and is the lexicographic order of the spatial weights and **not** the order of the input.

**Comment by Michel-Anton** [comment]
Thank you for the quick answer. I took a look at pysal/libpysal#184. This is a significant change in the structure of libpysal and downstream repercussions are hard to foresee.

The reordering of `adj_list_gs` is the natural fix. I believe the same fix should be applied to geary_local_mv.py because the code lines 113-116 induce a similar loss of order:
```
temp = pd.DataFrame(gs).T
temp['ID'] = adj_list.focal.values
adj_list_gs = temp.groupby(by='ID').sum()
localG = np.array(adj_list_gs.sum(axis=1) / k)
```

**Comment by ljwolf** [comment]
OK, this should be addressed in #195. For example:

```python
import libpysal, geopandas, numpy, esda

guerry = geopandas.read_file("./guerry.shp")

guerry_shuffled = guerry.sample(frac=1, replace=False)
guerry_abet = guerry.sort_values('Dprtmnt')

w = libpysal.weights.Queen.from_dataframe(guerry)
w_shuffled = libpysal.weights.Queen.from_dataframe(guerry_shuffled)
w_abet = libpysal.weights.Queen.from_dataframe(guerry, ids='Dprtmnt')

lG = esda.Geary_Local(connectivity=w).fit(guerry['Donatns'])
lG_shuffled = esda.Geary_Local(connectivity=w_shuffled).fit(guerry_shuffled['Donatns'])
lG_abet =  esda.Geary_Local(connectivity=w_abet).fit(guerry_abet['Donatns'])

guerry['localG'] = lG.localG
guerry_shuffled['localG'] = lG_shuffled.localG
guerry_abet['localG'] = lG_abet.localG

numpy.testing.assert_allclose(guerry.sort_values('Dprtmnt').localG, guerry_abet.localG)
numpy.testing.assert_allclose(guerry_shuffled.sort_values('Dprtmnt').localG, guerry_abet.localG)
```

This indicates that the "right" `localG` score now corresponds to the "right" department, even when input weights ids are out of order (integers or set explicitly as strings). Still note, though, the silent lexical sorting induced for `w_abet`, which we need to change in libpysal#184, and needs @sjsrey approval plus (I think) a major version increment!
